evaluator/builtin_string.go

```
@@ -637,3 +639,94 @@ func builtinRpad(args []types.Datum, ctx context.Context) (d types.Datum, err er
 
 	return d, nil
 }
+
+// https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_bit-length
+func builtinBitLength(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
+		return d, nil
+	}
+
+	str, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	d.SetInt64(int64(len(str) * 8))
+	return d, nil
+}
+
+// https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_char
+func builtinChar(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	// The kinds of args are int or string, and the last one represents charset.
+	result := make([]byte, 0, len(args)*2)
+	var resultLen int
+	var datumInt int64
+
+	for _, datum := range args[:len(args)-1] {
+		switch datum.Kind() {
+		case types.KindNull:
+			continue
+		case types.KindString:
+			i, err := datum.ToInt64(ctx.GetSessionVars().StmtCtx)
+			if err != nil {
+				d.SetString("")
+				return d, nil
+			}
+			datumInt = i
+		case types.KindInt64, types.KindUint64, types.KindMysqlHex, types.KindFloat32, types.KindFloat64, types.KindMysqlDecimal:
+			x, _ := datum.Cast(ctx.GetSessionVars().StmtCtx, types.NewFieldType(mysql.TypeLonglong))
```

Don't ignore error....